linkis-computation-governance/linkis-manager/linkis-application-manager/src/main/scala/org/apache/linkis/manager/rm/external/yarn/YarnResourceRequester.scala

```diff
@@ -77,31 +73,37 @@ class YarnResourceRequester extends ExternalResourceRequester with Logging {
     logger.info(s"rmWebAddress: $rmWebAddress")
     val queueName = identifier.asInstanceOf[YarnResourceIdentifier].getQueueName
 
-    def getYarnResource(jValue: Option[JValue]) = jValue.map(r =>
-      new YarnResource(
-        (r \ "memory").asInstanceOf[JInt].values.toLong * 1024L * 1024L,
-        (r \ "vCores").asInstanceOf[JInt].values.toInt,
-        0,
-        queueName
-      )
-    )
+    def getYarnResource(resource: Option[Any]) = resource.map(r => {
+      val value =
+        JsonUtils.jackson.readValue(r.asInstanceOf[YarnResource].toJson, classOf[Map[String, Any]])
+      val memory = getMapValue[Int](value, "memory")
+      val vCores = getMapValue[Int](value, "vCores")
+      new YarnResource(memory.toLong * 1024L * 1024L, vCores, 0, queueName)
+    })
 
-    def maxEffectiveHandle(queueValue: Option[JValue]): Option[YarnResource] = {
+    def maxEffectiveHandle(queueValue: Option[Any]): Option[YarnResource] = {
       val metrics = getResponseByUrl("metrics", rmWebAddress)
-      val totalResouceInfoResponse = (
-        (metrics \ "clusterMetrics" \ "totalMB").asInstanceOf[JInt].values.toLong,
-        (metrics \ "clusterMetrics" \ "totalVirtualCores").asInstanceOf[JInt].values.toLong
-      )
+      val metricsJson = JsonUtils.jackson.readValue(metrics.toString(), classOf[Map[String, Any]])
+      val clusterMetricsJson = getMapValue[String](metricsJson, "clusterMetrics")
+
+      val clusterMetrics =
+        JsonUtils.jackson.readValue(clusterMetricsJson, classOf[Map[String, Any]])
+      val totalMB = getMapValue[Int](clusterMetrics, "totalMB")
```

Comment:
this can use jsonPath to replace?

Reply:
and all other locations of get a spec path json value

Reply:
goog idea, will try it later